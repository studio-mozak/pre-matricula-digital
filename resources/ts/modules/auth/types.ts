export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
