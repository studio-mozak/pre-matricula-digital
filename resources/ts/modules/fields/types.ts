import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
