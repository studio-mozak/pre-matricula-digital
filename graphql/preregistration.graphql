extend type Query @guard(with: ["prematricula"]) {

  """
  Lista as pré-matrículas.
  """
  preregistrations(
    """
    Retorna apenas as pré-matrículas onde o **protocolo** ou o **nome do(a) aluno(a)** contenha `search`.
    """
    search: String @slug @search(columns: ["protocol", "student.slug"])
    process: ID @eq(key: "process_id")
    processes: [ID!] @in(key: "process_id")
    school: ID @eq(key: "school_id")
    schools: [ID!] @in(key: "school_id")
    grade: ID @eq(key: "grade_id")
    grades: [ID] @in(key: "grade_id")
    period: ID @eq(key: "period_id")
    type: PreRegistrationType @eq(key: "preregistration_type_id")
    status: PreRegistrationStatus @eq(key: "status")
    sort: PreRegistrationSort @scope(name: "sort")
    year: Int @search(columns: ["process.school_year_id"])
  ): [PreRegistration!]! @paginate

}

extend type Query @guard(with: ["web", "sanctum", "prematricula"]) {

  """
  Retorna a pré-matrícula do protocolo.
  """
  preregistrationByProtocol(
    protocol: String @eq
    withVacancy: Boolean
  ): PreRegistration @first

  """
  Retorna a inscrição usando o código de verificação.
  """
  preregistrationByCode(
    code: String @eq
  ): PreRegistration @first

  getStudentMatches(
    stage: ID!
    name: String @slug
    dateOfBirth: Date @rename(attribute: "date_of_birth")
    cpf: String
    rg: String
    birthCertificate: String @rename(attribute: "birth_certificate")
  ): [StudentMatch!]!

  summonNextInLine(
    process: ID! @eq(key: "process_id")
    school: ID! @eq(key: "school_id")
    grade: ID! @eq(key: "grade_id")
    period: ID! @eq(key: "period_id")
  ): PreRegistration @first(scopes: ["nextInLine", "waiting"])

  matchByStudentCpf(
    cpf: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentRg(
    rg: String! @eq
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentBirthCertificate(
    birthCertificate: String! @eq(key: "birth_certificate")
    year: Int! @eq
  ): RegistrationMatch @first

  matchByStudentNameAndDateOfBirth(
    name: String! @slug @eq(key: "slug")
    dateOfBirth: Date! @eq(key: "date_of_birth")
    year: Int! @eq
  ): RegistrationMatch @first

  findPreRegistrationsByStudentCpf(
    cpf: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentRg(
    rg: String!
  ): [PreRegistration!]!

  findPreRegistrationsByStudentBirthCertificate(
    birthCertificate: String! @rename(attribute: "birth_certificate")
  ): [PreRegistration!]!

  findPreRegistrationsByStudentNameAndDateOfBirth(
    name: String! @slug
    dateOfBirth: Date! @rename(attribute: "date_of_birth")
  ): [PreRegistration!]!

  findPreRegistrationsByInfo(
    cpf: String @rename(attribute: "responsible_cpf")
    email: String @rename(attribute: "responsible_email")
    protocol: String
  ): [PreRegistration!]!

  getPreRegistrationLinkedByEmail(
    email: String! @eq
  ): [PreRegistrationLinkedByEmail!]! @all

  searchLegacyStudentsByName(
    name: String!
    first: Int
  ): [LegacyStudent!]!
}

enum PreRegistrationType {
  """
  Rematrícula
  """
  REGISTRATION_RENEWAL @enum(value: 1)
  """
  Matrícula
  """
  REGISTRATION @enum(value: 2)
  """
  Lista de espera
  """
  WAITING_LIST @enum(value: 3)
}

enum PreRegistrationStatus {
  """
  Pré-matrícula aguardando (in)deferimento.
  """
  WAITING @enum(value: 1)

  """
  Pré-matrícula deferida.
  """
  ACCEPTED @enum(value: 2)

  """
  Pré-matrícula indeferida.
  """
  REJECTED @enum(value: 3)

  """
  Pré-matrícula convocada.
  """
  SUMMONED @enum(value: 4)

  """
  Pré-matrícula em confirmação.
  """
  IN_CONFIRMATION @enum(value: 5)
}

enum RelationType {
  """
  Mãe
  """
  MOTHER @enum(value: 1)

  """
  Pai
  """
  FATHER @enum(value: 2)

  """
  Responsável legal
  """
  GUARDIAN @enum(value: 3)

  """
  A própria pessoa
  """
  SELF @enum(value: 4)
}

enum PreRegistrationSort {
  DATE @enum(value: 1)
  POSITION @enum(value: 2)
  SCHOOL @enum(value: 3)
  NAME @enum(value: 4)
  DATE_OF_BIRTH @enum(value: 5)
}

type RegistrationMatch {
  id: ID!
  year: Int!
  school: ID!
  grade: ID!
  period: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
}

enum StudentMatchType {
  PRE_REGISTRATION_ALREADY_EXISTS @enum(value: 1)
  REGISTRATION_CURRENT_YEAR @enum(value: 2)
  REGISTRATION_LAST_YEAR @enum(value: 3)
  STUDENT_ALREADY_EXISTS @enum(value: 4)
}

type StudentMatch {
  id: ID!
  initials: String!
  dateOfBirth: Date! @rename(attribute: "date_of_birth")
  type: ProcessStageRestriction!
  process: Process
  registration: StudentMatchRegistration
}

type StudentMatchRegistration {
  year: Int!
  school: School!
  grade: Grade!
  period: Period
}

type PreRegistration {
  id: ID!
  type: PreRegistrationType! @rename(attribute: "preregistration_type_id")
  position: Int!
  parent: PreRegistration @belongsTo
  waiting: PreRegistration @belongsTo
  others: [PreRegistration]
  protocol: String!
  code: String!
  date: DateTime! @rename(attribute: "created_at")
  process: Process @belongsTo
  period: Period @belongsTo
  school: School @belongsTo
  grade: Grade @belongsTo
  schoolYear: SchoolYear @belongsTo
  classroom: Classroom @belongsTo
  inClassroom: Classroom @belongsTo
  student: Person @belongsTo
  responsible: Person @belongsTo
  relationType: RelationType! @rename(attribute: "relation_type_id")
  status: PreRegistrationStatus!
  observation: String
  fields: [PreRegistrationField!]! @hasMany
  stage: ProcessStage @belongsTo
  external: ExternalPerson @hasOne
  linked: [PreRegistrationLinkedByEmail!]! @hasMany
}

type PreRegistrationField {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  field: Field! @belongsTo
  value: String
}

type PreRegistrationLinkedByEmail {
  id: ID!
  preregistration: PreRegistration! @belongsTo
  email: String!
}

extend type Mutation @guard(with: ["web", "sanctum", "prematricula"]) {

  newPreRegistration(
    input: PreRegistrationInput! @spread
  ): [PreRegistration!]!

  sendProtocolsByEmail(
    preregistrations: [ID!]!
    email: String!
  ): Boolean!

  keepOnTheWaitingList(
    id: ID!
    grade: ID
  ): Boolean!

  updateStudentInExternalSystem(
    preregistration: ID!
    cpf: Boolean
    rg: Boolean
    birthCertificate: Boolean
    name: Boolean
    dateOfBirth: Boolean
    gender: Boolean
    phone: Boolean
    mobile: Boolean
    address: Boolean
  ): Boolean!

  savePreRegistrationLinkedByEmail(
    preregistration: ID! @rename(attribute: "preregistration_id")
    email: String!
  ): PreRegistrationLinkedByEmail!

  deletePreRegistrationLinkedByEmail(
    id: ID! @whereKey
  ): PreRegistrationLinkedByEmail! @delete

}

extend type Mutation @guard {
  acceptPreRegistrations(
    ids: [ID!]!
    classroom: ID!
  ): [PreRegistration!]!

  rejectPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  summonPreRegistrations(
    ids: [ID!]!
    justification: String
  ): [PreRegistration!]!

  returnToWaitPreRegistrations(
    ids: [ID!]!
  ): [PreRegistration!]!

  updateStudent(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updateResponsible(
    protocol: String!
    fields: [PreRegistrationFieldInput!]!
  ): PreRegistration

  updatePreRegistration(
    protocol: String!
    grade: ID!
    school: ID!
    period: ID!
  ): PreRegistration

  updateAddress(
    protocol: String!
    address: AddressInput
  ): PreRegistration
}

input PreRegistrationFieldInput {
  field: ID!
  value: String
}

input AddressInput {
  postalCode: String! @rename(attribute: "postal_code")
  address: String!
  number: String!
  complement: String
  neighborhood: String!
  city: String
  stateAbbreviation: String
  cityIbgeCode: Int
  lat: Float! @rename(attribute: "latitude")
  lng: Float! @rename(attribute: "longitude")
  manualChangeLocation: Boolean! @rename(attribute: "manual_change_location")
}

input PreRegistrationWaitingListInput {
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
}

input PreRegistrationInput {
  type: PreRegistrationType!  @rename(attribute: "preregistration_type_id")
  process: ID! @rename(attribute: "process_id")
  stage: ID! @rename(attribute: "process_stage_id")
  grade: ID! @rename(attribute: "grade_id")
  period: ID! @rename(attribute: "period_id")
  school: ID @rename(attribute: "school_id")
  optionalSchool: ID
  optionalPeriod: ID
  address: AddressInput!
  optionalAddress: AddressInput
  relationType: RelationType! @rename(attribute: "relation_type_id")
  responsible: [PreRegistrationFieldInput!]!
  student: [PreRegistrationFieldInput!]!
  externalPerson: ID @rename(attribute: "external_person_id")
  waitingList: [PreRegistrationWaitingListInput!]
}

type LegacyStudent {
  id: ID!
  name: String!
  cpf: String
  dateOfBirth: String @rename(attribute: "date_of_birth")
  responsibleName: String @rename(attribute: "responsible_name")
  responsibleCpf: String @rename(attribute: "responsible_cpf")
  responsibleDateOfBirth: String @rename(attribute: "responsible_date_of_birth")
  responsiblePhone: String @rename(attribute: "responsible_phone")
}
