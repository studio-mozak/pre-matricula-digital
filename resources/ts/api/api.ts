import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
