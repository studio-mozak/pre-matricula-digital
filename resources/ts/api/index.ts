export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
