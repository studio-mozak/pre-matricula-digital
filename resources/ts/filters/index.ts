import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
