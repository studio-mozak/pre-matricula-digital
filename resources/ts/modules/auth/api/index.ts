/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
