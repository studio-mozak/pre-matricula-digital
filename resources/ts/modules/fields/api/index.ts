/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
