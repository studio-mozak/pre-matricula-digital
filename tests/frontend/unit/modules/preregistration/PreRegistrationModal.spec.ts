import '@/plugin/sortby';
import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { expect, vi } from 'vitest';
import { useRoute, useRouter } from 'vue-router';
import { Option } from '@/types';
import { PreRegistration } from '@/modules/preregistration/types';
import PreRegistrationModal from '@/modules/preregistration/components/PreRegistrationModal.vue';
import { Preregistration as PreregistrationApi } from '@/modules/preregistration/api';
import VTooltip from 'v-tooltip';
import { createTestingPinia } from '@pinia/testing';
import filters from '@/filters';

vi.mock('@/modules/preregistration/api');
vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
  useRoute: () => ({
    params: {
      protocol: 'HLYN3T',
    },
  }),
}));

const spyPush = vi.fn();

const loaderByProtocolMock: PreRegistration = {
  id: 1,
  date: '2020-11-17 00:02:40',
  protocol: 'HLYN3T',
  status: 'REJECTED',
  type: 'REGISTRATION',
  observation: '',
  grade: {
    id: '1',
    name: '1º Ano',
    course: {
      name: 'Ensino Fundamental 9 anos',
    },
  },
  period: {
    id: '1',
    name: 'Vespertino',
  },
  school: {
    id: '1',
    name: 'Escola 001',
  },
  classroom: {
    id: '1',
    name: '67',
  },
  position: 4,
  waiting: {
    id: '1',
    protocol: 'HLYN3T',
    school: {
      id: '1',
      name: 'Escola 001',
    },
    position: 4,
  },
  parent: {
    id: '1',
    protocol: 'HLYN3T',
    school: {
      id: '1',
      name: 'Escola 001',
    },
    position: 4,
  },
  student: {
    student_birth_certificate: 'null',
    student_cpf: 'null',
    student_date_of_birth: '2014-12-06',
    student_email: 'null',
    student_gender: 1,
    student_marital_status: 1,
    student_mobile: 'null',
    student_name: 'Daenerys Targarien',
    student_phone: 'null',
    student_place_of_birth: 4,
    student_rg: 'null',
  },
  responsible: {
    responsible_address: [],
    responsible_cpf: '107.247.559-69',
    responsible_date_of_birth: '1996-06-03',
    responsible_email: 'null',
    responsible_gender: 1,
    responsible_marital_status: 1,
    responsible_mobile: 'null',
    responsible_name: 'Jhon Snow',
    responsible_phone: '(99) 99999-9999',
    responsible_place_of_birth: 2,
    responsible_rg: 'null',
  },
  relationType: 'MOTHER',
  fields: [
    {
      id: '1',
      value: '18',
      field: {
        id: '19',
        name: 'Aluno possui algum tipo de deficiência/ Síndrome?',
        internal: 'null',
        group: 'STUDENT',
      },
    },
    {
      id: '2',
      value: '18',
      field: {
        id: '20',
        name: 'Aluno possui algum tipo de deficiência/ Síndrome?',
        internal: 'null',
        group: 'RESPONSIBLE',
      },
    },
  ],
  process: {
    id: '1',
    name: 'Ensino Fundamental (1º ao 9º ano)',
    fields: [
      {
        id: '13',
        order: 7,
        field: {
          id: 'string',
          name: 'string',
          group: 'string',
          internal: 'responsible_rg',
          type: 'GENDER',
          options: [
            {
              id: 'string',
              key: 'string',
              name: 'string',
              label: 'string',
              weight: 0,
            },
          ],
        },
        required: true,
        weight: 0,
      },
    ],
    schoolYear: {
      year: 20211,
    },
  },
  inClassroom: null,
  others: [],
};

const getClassroomsByPreregistrationMock: Option[] = [
  {
    key: 1,
    label: 'string',
  },
];

const onMounted = async (
  loaderByProtocol = loaderByProtocolMock,
  getClassroomsByPreregistration = getClassroomsByPreregistrationMock
) => {
  vi.mocked(PreregistrationApi.listByProtocol).mockResolvedValueOnce(
    loaderByProtocol
  );
  vi.mocked(
    PreregistrationApi.getClassroomsByPreregistration
  ).mockResolvedValueOnce(getClassroomsByPreregistration);

  const wrapper = mount(PreRegistrationModal, {
    global: {
      provide: {
        $filters: filters,
      },
      plugins: [createTestingPinia()],
      stubs: ['x-field'],
      directives: {
        tooltip: VTooltip,
      },
    },
  });

  await wrapper.vm.$nextTick();
  await flushPromises();

  return wrapper;
};

describe('PreRegistrationModal', () => {
  let wrapper: VueWrapper<InstanceType<typeof PreRegistrationModal>>;

  beforeEach(async () => {
    wrapper = await onMounted(loaderByProtocolMock);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('accept', () => {
    test('should return with success response', async () => {
      vi.mocked(PreregistrationApi.postAccept).mockResolvedValueOnce(undefined);

      wrapper.vm.step = 'ACCEPT';
      wrapper.vm.classroom = 123;
      wrapper.vm.accept();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(spyPush).toHaveBeenCalledTimes(1);
      expect(spyPush).toHaveBeenCalledWith({
        name: 'preregistrations',
      });
    });

    test('should return with fail response', async () => {
      const errors = [
        {
          message: 'string',
          extensions: {
            message: 'string',
          },
        },
      ];

      vi.mocked(PreregistrationApi.postAccept).mockRejectedValueOnce({
        response: {
          data: {
            errors,
          },
        },
      });

      wrapper.vm.step = 'ACCEPT';
      wrapper.vm.classroom = 123;
      wrapper.vm.accept();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(spyPush).not.toHaveBeenCalled();
      expect(wrapper.vm.showError).toBeTruthy();
      expect(wrapper.vm.errorTitle).toBe(errors[0].message);
      expect(wrapper.vm.error).toBe(errors[0].extensions.message);
    });

    test('should show error message when classroom is undefined', async () => {
      const errors = [
        {
          message: 'string',
          extensions: {
            message: 'string',
          },
        },
      ];

      vi.mocked(PreregistrationApi.postAccept).mockRejectedValueOnce({
        response: {
          data: {
            errors,
          },
        },
      });

      wrapper.vm.step = 'ACCEPT';
      wrapper.vm.classroom = undefined;
      wrapper.vm.accept();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(spyPush).not.toHaveBeenCalled();
      expect(wrapper.vm.showError).toBeTruthy();
      expect(wrapper.vm.errorTitle).toBe(
        'É necessário selecionar uma turma para realizar o Deferimento da pré-matrícula.'
      );
      expect(wrapper.vm.error).toBe('');
    });
  });

  describe('summon', () => {
    test('should return with success response', async () => {
      vi.mocked(PreregistrationApi.postSummon).mockResolvedValueOnce(undefined);

      wrapper.vm.step = 'SUMMON';
      wrapper.vm.summon();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(spyPush).toHaveBeenCalledTimes(1);
      expect(spyPush).toHaveBeenCalledWith({
        name: 'preregistrations',
      });
    });

    test('should return with fail response', async () => {
      const errors = [
        {
          message: 'string',
          extensions: {
            message: 'string',
          },
        },
      ];

      vi.mocked(PreregistrationApi.postSummon).mockRejectedValueOnce({
        response: {
          data: {
            errors,
          },
        },
      });

      wrapper.vm.step = 'SUMMON';
      wrapper.vm.summon();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(spyPush).not.toHaveBeenCalled();
      expect(wrapper.vm.showError).toBeTruthy();
      expect(wrapper.vm.errorTitle).toBe(
        'Não foi possível realizar a convocação de todos os(as) responsáveis(as)'
      );
      expect(wrapper.vm.error).toBe(errors[0].extensions.message);
    });
  });

  describe('go', () => {
    test('should return redirect', () => {
      wrapper.vm.go('HLYN3T');

      expect(spyPush).toHaveBeenCalledWith({
        name: 'preregistration.modal',
        params: {
          protocol: 'HLYN3T',
        },
      });
    });
  });

  describe('getClassroomsByPreregistration', () => {
    test('should call the getClassroomsByPreregistration method with the correct params', async () => {
      vi.mocked(
        PreregistrationApi.getClassroomsByPreregistration
      ).mockResolvedValueOnce(getClassroomsByPreregistrationMock);
      wrapper.vm.getClassroomsByPreregistration();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(wrapper.vm.classrooms).toMatchObject(
        getClassroomsByPreregistrationMock
      );
    });
  });

  describe('returnToWait', () => {
    test('should call the returnToWait method with the correct params', async () => {
      vi.mocked(PreregistrationApi.postReturnToWait).mockResolvedValueOnce(
        undefined
      );

      wrapper.vm.returnToWait();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(spyPush).toHaveBeenCalledWith({
        name: 'preregistrations',
      });
    });
  });

  describe('reject', () => {
    test('should call the reject method with the correct params', async () => {
      vi.mocked(PreregistrationApi.postReject).mockResolvedValueOnce(undefined);
      wrapper.vm.step = 'REJECT';
      wrapper.vm.reject();

      await wrapper.vm.$nextTick();
      await flushPromises();

      expect(spyPush).toHaveBeenCalledWith({
        name: 'preregistrations',
      });
    });

    test('should return justification empty string', () => {
      wrapper.vm.reject();
      wrapper.vm.step = 'SHOW';
      expect(wrapper.vm.justification).toBe('');
    });
  });
});
