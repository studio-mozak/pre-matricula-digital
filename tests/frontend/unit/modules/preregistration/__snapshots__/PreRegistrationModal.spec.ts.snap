// Vitest Snapshot v1

exports[`PreRegistrationModal > should match snapshot 1`] = `<!--teleport start-->`;
